helix-lock/src/main/java/org/apache/helix/lock/HelixLock.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.helix.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Generic interface for Helix distributed lock
+ */
+public interface HelixLock {
+  /**
+   * Acquire a lock
```

Comment:
The interface is for both blocking or nonblocking calls. I added into the comment.

Reply:
If we design the method for both types, it may end up with only one implementation at a time. What if user may have both use cases for same lock object?

Reply:
@dasahcc Not sure I fully understand the scenario you described. Can you provide a specific example? What we want to achieve is to hide the implementation from our customers. So they choose to use nonblocking or blocking lock by using different HelixLock constructors, then they use the lock by calling `acquire` or `release`, we handle the difference of nonblocking and blocking in implementations.

Reply:
Currently this interface only includes blocking calls. Nonblocking calls can be added later if there is a need from our customers.